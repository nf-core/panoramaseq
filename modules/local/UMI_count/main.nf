process UMI_count {
    tag "$meta.id"
    label "process_long"

    // conda "${moduleDir}/environment.yml"
    
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/umi_tools:1.1.5--py39hf95cd2a_0' :
        'biocontainers/umi_tools:1.1.5--py39hf95cd2a_0' }"
    
    input:
    tuple val(meta), path(bam), path(bam_index)
    
    output:
    tuple val(meta), path("*.tsv.gz"), emit: umi_counts
    path('logs_umi/*.{txt,log}'), emit: log_files
    path "versions.yml", emit: versions

    script:
    """
    mkdir logs_umi
    umi_tools count --per-gene --gene-tag=XT --assigned-status-tag=XS --per-cell -I ${bam} -S ${meta.id}_counts.tsv.gz --log=logs_umi/${meta.id}_umiCounts.log
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        umi_tools: \$(umi_tools --version | sed '/version:/!d; s/.*: //')
END_VERSIONS
    """

    stub:
    """
    mkdir logs_umi
    touch ${meta.id}_counts.tsv.gz
    touch logs_umi/${meta.id}_umiCounts.log
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        umi_tools: "stub-version"
END_VERSIONS
    """

}