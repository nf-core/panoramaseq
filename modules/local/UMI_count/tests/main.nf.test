nextflow_process {

    name "Test Process UMI_count"
    script "../main.nf"
    process "UMI_count"
    config "./nextflow.config"
    tag "modules"
    tag "modules_local"
    tag "umi_count"

    test("panoramaseq [bam] - S09") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'S09', N_barcodes:34500, sample_size:10000, len_barcode:36, Nthresh:9, Ntriage:1000 ], // meta map
                    file('https://raw.githubusercontent.com/francops1722/test-datasets/panoramaseq/testdata/samtools/S09.assigned_sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/francops1722/test-datasets/panoramaseq/testdata/samtools/S09.assigned_sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("panoramaseq [bam] - S10") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'S10', N_barcodes:34500, sample_size:10000, len_barcode:36, Nthresh:9, Ntriage:1000 ], // meta map
                    file('https://raw.githubusercontent.com/francops1722/test-datasets/panoramaseq/testdata/samtools/S10.assigned_sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/francops1722/test-datasets/panoramaseq/testdata/samtools/S10.assigned_sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("panoramaseq [bam] - S09 - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'S09', N_barcodes:34500, sample_size:10000, len_barcode:36, Nthresh:9, Ntriage:1000 ], // meta map
                    file('https://raw.githubusercontent.com/francops1722/test-datasets/panoramaseq/testdata/samtools/S09.assigned_sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/francops1722/test-datasets/panoramaseq/testdata/samtools/S09.assigned_sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
