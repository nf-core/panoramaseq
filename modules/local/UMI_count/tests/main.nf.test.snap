{
    "panoramaseq [bam] - S09": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "S09",
                            "N_barcodes": 34500,
                            "sample_size": 10000,
                            "len_barcode": 36,
                            "Nthresh": 9,
                            "Ntriage": 1000
                        },
                        "S09_counts.tsv.gz:md5,5f1f7122a243a38d9e177f0836cf46cf"
                    ]
                ],
                "1": [
                    "S09_umiCounts.log:md5,4774d4f545cb793b38daf25197d6ba9d"
                ],
                "2": [
                    "versions.yml:md5,1e4c6449d6ecada2d47a94b0ecdaf8b3"
                ],
                "log_files": [
                    "S09_umiCounts.log:md5,4774d4f545cb793b38daf25197d6ba9d"
                ],
                "umi_counts": [
                    [
                        {
                            "id": "S09",
                            "N_barcodes": 34500,
                            "sample_size": 10000,
                            "len_barcode": 36,
                            "Nthresh": 9,
                            "Ntriage": 1000
                        },
                        "S09_counts.tsv.gz:md5,5f1f7122a243a38d9e177f0836cf46cf"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1e4c6449d6ecada2d47a94b0ecdaf8b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T16:37:15.51068879"
    },
    "panoramaseq [bam] - S09 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "S09",
                            "N_barcodes": 34500,
                            "sample_size": 10000,
                            "len_barcode": 36,
                            "Nthresh": 9,
                            "Ntriage": 1000
                        },
                        "/kyukon/scratch/gent/vo/000/gvo00027/vsc44685/panoramaseq/nf-core-panoramaseq/.nf-test/tests/ed4300186a8d5bb05d295f3e64563ea6/work/fb/f620df6e36c97d3447c508d86cf9da/S09_counts.tsv.gz"
                    ]
                ],
                "1": [
                    "S09_umiCounts.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,d07c692167ea33064616e3dd2d41bd69"
                ],
                "log_files": [
                    "S09_umiCounts.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "umi_counts": [
                    [
                        {
                            "id": "S09",
                            "N_barcodes": 34500,
                            "sample_size": 10000,
                            "len_barcode": 36,
                            "Nthresh": 9,
                            "Ntriage": 1000
                        },
                        "/kyukon/scratch/gent/vo/000/gvo00027/vsc44685/panoramaseq/nf-core-panoramaseq/.nf-test/tests/ed4300186a8d5bb05d295f3e64563ea6/work/fb/f620df6e36c97d3447c508d86cf9da/S09_counts.tsv.gz"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d07c692167ea33064616e3dd2d41bd69"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T16:37:31.275951746"
    },
    "panoramaseq [bam] - S10": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "S10",
                            "N_barcodes": 34500,
                            "sample_size": 10000,
                            "len_barcode": 36,
                            "Nthresh": 9,
                            "Ntriage": 1000
                        },
                        "S10_counts.tsv.gz:md5,2a406a1644cc927b06bc27b5e1704e83"
                    ]
                ],
                "1": [
                    "S10_umiCounts.log:md5,ddb22a7d689a54fe52641e9e1a86eda4"
                ],
                "2": [
                    "versions.yml:md5,1e4c6449d6ecada2d47a94b0ecdaf8b3"
                ],
                "log_files": [
                    "S10_umiCounts.log:md5,ddb22a7d689a54fe52641e9e1a86eda4"
                ],
                "umi_counts": [
                    [
                        {
                            "id": "S10",
                            "N_barcodes": 34500,
                            "sample_size": 10000,
                            "len_barcode": 36,
                            "Nthresh": 9,
                            "Ntriage": 1000
                        },
                        "S10_counts.tsv.gz:md5,2a406a1644cc927b06bc27b5e1704e83"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1e4c6449d6ecada2d47a94b0ecdaf8b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T16:37:25.778164385"
    }
}